refactor export test vectors

chee-chyuan opened the pull request:
https://github.com/o1-labs/proof-systems/issues/333

chee-chyuan commented:
hi @mimoo lmk if im doing this correctly. thanks

mimoo commented:
Awesome, thanks for the PR! Could you actually move this crate inside of oracle/ ? I don’t think it makes sense to have it in the root folder

mimoo commented on export_test_vectors/Cargo.toml:
```diff
@@ -0,0 +1,21 @@
+oracle = { path = "../oracle" }
+
+[features]
+test_vectors = ["num-bigint", "serde_json", "hex", "ark-serialize"]
```
missing new line at the end

chee-chyuan commented:
hi @mimoo thanks. made a few more changes which mainly involves moving `export_test_vectors` back to `oracle` and adding it as a `workspace`. 

also removed the `test_vector` features and added features in the `default` in its `Cargo.toml`. as a result we can remove the `--all-features` flag. lmk if this is what is expected.

mimoo commented on oracle/README.md:
````diff
@@ -7,13 +7,13 @@ Tests are ran against our own test vectors in `tests/test_vectors`.
 
 ```text
-usage: cargo run --all-features -- [Hex|B10] [legacy|kimchi] <OUTPUT_FILE>
+usage: cargo run -p export_test_vectors -- [Hex|B10] [legacy|kimchi] <OUTPUT_FILE>
````
can you remove `usage:` here? I know you didn't add it, don't know why it was there in the first place :o

mimoo commented on oracle/README.md:
```diff
@@ -7,13 +7,13 @@ Tests are ran against our own test vectors in `tests/test_vectors`.
-cargo run --all-features -- hex kimchi kimchi.json
+cargo run -p export_test_vectors -- B10 legacy -
+cargo run -p export_test_vectors -- b10 legacy legacy.json
+cargo run -p export_test_vectors -- hex kimchi kimchi.json
```
looks like it works without the `-p` option as well (probably because there's only one binary)

chee-chyuan replied:
i tried running in the oracle folder directly and its complaining when i didnt have `-p` .  however it works well without `-p` when i run in the `export_test_vectors` folder. wondering if you are seeing the same too

mimoo replied:
oh wait I'm stupid, you're right!

mimoo commented on oracle/export_test_vectors/Cargo.toml:
```diff
@@ -0,0 +1,21 @@
+oracle = { path = "../../oracle" }
+
+[features]
+default = ["num-bigint", "serde_json", "hex", "ark-serialize"]
```
I don't think we need any features here, so you can just all the `optional = true`

mimoo commented on oracle/export_test_vectors/src/main.rs:
```diff
@@ -1,17 +1,11 @@
-#[cfg(feature = "test_vectors")]
 mod vectors;
-#[cfg(feature = "test_vectors")]
```
great!

mimoo reviewed (commented):
This is great! This works :) I just left you a few nits, but otherwise it looks good to me!

mimoo reviewed (approved):
Thanks a lot for this! I'll merge it once the tests pass but it looks good to me now :)

Let me know if you're interested in doing another task!

chee-chyuan commented:
> Thanks a lot for this! I'll merge it once the tests pass but it looks good to me now :)
> 
> Let me know if you're interested in doing another task!

im definitely up for another task. hit me up
